### PR TITLE
Fix saving search list

### PR DESCRIPTION
**Pull request by stanleylhs**
Updated model with optional and controller saving with attr

### CODE REVIEW

File: app/controllers/lists_controller.rb [hunk 1]
```diff
@@ -9,13 +9,14 @@ class ListsController < ApplicationController
   # POST /lists
   #----------------------------------------------------------------------------
   def create
-    list_params[:user_id] = (current_user.id if params[:is_global].to_i.zero?)
+    list_attr = list_params.to_h
+    list_attr[:user_id] = (current_user.id if params[:is_global].t o_i.zero?)
```
Comment (CloCkWeRX) [inline app/controllers/lists_controller.rb:13]:
```suggestion
    list_attr[:user_id] = (current_user.id if params[:is_global].to_i.zero?)
```

**Comment by CloCkWeRX** [comment]
This could do with a test to show the bug

**Comment by stanleylhs** [comment]
> This could do with a test to show the bug

Sorry my bad, I must have put a white space there by mistake, let me just amend it.

File: app/controllers/lists_controller.rb (hunk 1 above)
Comment (stanleylhs) [reply app/controllers/lists_controller.rb]: Sorry my bad, I must have put a white space there by mistake, let me just amend it.

**Comment by steveyken** [comment]
We were missing some specs for ListController so I've added those now.

**Review by steveyken — APPROVED** [review]

**Pull request merged**